### PR TITLE
Update docs for v1

### LINKED ISSUE #647: Update docs to reflect Configu v1
### Suggestion

The documentation website does not reflect the changes made to the project and the interfaces as part of the release of Configu v1. 

List of pages identified as needing an update:
- https://docs.configu.com/introduction/overview
  - The quickstart/interfaces accordion -> interfaces -> internal SDKs -> the copy should be updated
- https://docs.configu.com/introduction/concepts
  - `Cfgu` - the copy needs to be updated, `type` is no longer a thing
  - `ConfigStore` - `ConfigStore Collection` seems like a broken link
  - `learn-more-about-configstore`
    - `.configu file` - broken link
    - `support a wide range of ConfigStores` - broken link
  - `ConfigSchema` - `learn-more-about-configschema`
    - The second paragraph mentions the `type` cfgu prop which doesn't exist anymore
    - The examples of schema files are outdated in terms of file structure
  - `Sercert Manager` - typo error, should be `Secret Manager`
  - Add a section for `ConfigExpression`
- https://docs.configu.com/introduction/workflow
  - The image shows an outdated schema file structure and the old `type` property
  - `Export Configurations` - `Purpose` - outdated copy which mentions `Configu Proxy & SDKs`, quite sure `SDKs` is wrong now.
- https://docs.configu.com/interfaces/.cfgu
  - The `Properties` section needs to be updated
  - The `Example` section needs to be updated
- https://docs.configu.com/interfaces/.configu
    - The opening paragraph says that `.configu` files are optional which
    - `Properties` - `stores` - change ref of `cache` to `backup`
    - `Properties` - `cache` - change to `backup`
    - `Example` - Change all refs of `cache` to `backup`
    - `environment-variables` ref to mustache should be removed as we don't use it anymore, we should just generically point to [configuration-template-engine](https://docs.configu.com/introduction/concepts#configuration-template-engine) without mentioning what engine we use here.
- https://docs.configu.com/interfaces/cli/setup
  - After finalizing cd-cli and installation scripts and decide on dns alias
- https://docs.configu.com/interfaces/cli/ref
  - Search for help dump for clipanion same as in oclif readme
- https://docs.configu.com/interfaces/proxy/ref
  - Update json schema for /export
  - Add more descriptions and styling
  - `GET /docs` - update image due to outdated schema
- https://docs.configu.com/interfaces/sdk/overview
  - Remove node, browser, and common
  - Rename ts to configu SDK
  - Enrich readme
  - Change tsdoc to produce sdk only docs
- #601
- Add docs for expressions under `@configu/integration`
- https://docs.configu.com/integrations/store/configu
    - NOTE: all these doc updates should only happen after the Configu Platform is migrated to v1. 
    - `Configs Explorer` - `Upsert Configs Wizard` - outdated schema structure in image
    - `Configs Editor` 
        - Old ref to `SDKs` that should be removed.
        - Editor form image shows old `type` cfgu prop
    - `Configs History`
        - `Evaluating with a tag` - the example needs to be updated
- https://docs.configu.com/guides/hashicorp-vault
    - outdated schema file structure
- #656

### Motivation

The docs must reflect the new state of the Configu Orchestrator to ensure users aren't onboarding with outdated information.

### PR DESCRIPTION
**Pull request by RichardAkman**
Closes #647

@rannn505, I need your help with:
- Updating the `Configu Flow Diagram` in `docs/introduction/workflow.mdx` to reflect the ConfigSchema structure. I don't have the source image.
- INPUT: I want to detail the context each `ConfigExpression` has depending on the relevant feature. I'm having some trouble deciding where to detail the context of all the CLI features that use expressions. The only place that makes sense is in the `ref.mdx` file but it's currently auto-generated. Should we maybe include the context details as part of the descriptions of each flag?
- I couldn't identify any tool to generate docs for `clipanion` similar to `oclif` readme. How should we proceed?

Notes:
- Also fixed some broken links in `CONTRIBUTING.md` and other files that still reference the old `@configu/ts` pacakge.
- Also fixed a broken link in `packages/integrations/stores/aws-secrets-manager/README.md`
- Also fixed a mistake in `packages/integrations/stores/toml-file/README.md`
- The entire Configu section of the docs shouldn't be updated until the platform has upgraded to v1. First, because the docs won't reflect reality until then. Second, the new screenshots can't be produced until the platform is updated.

### CODE REVIEW

File: docs/introduction/concepts.mdx [hunk 1]
```diff
@@ -214,28 +214,35 @@ Here is an example of a `ConfigSchema` file:
...
-In this example, the `ConfigSchema` binds a configuration called "API_KEY" with a type of "String". The configuration is marked as required and has a short description.
+In this example, the `ConfigSchema` binds a configuration called "API_KEY". The configuration is marked as required and has a short description.
 
 </Accordion>
 
+### ConfigExpression
+
+TODO:
+
+- Mention Acron and SES
```
Comment (RichardAkman) [inline docs/introduction/concepts.mdx:244]: Also mention all the inbuilt expressions like validator, lodash, etc...

File: docs/interfaces/cli/ref.mdx [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { Related } from '/snippets/callouts.mdx';
 <Tip>
   Use `configu -h` or to get a list of all available commands.
   <br />
+  Use `configu --config "path/to/.configu"` to explicitly provide a path to a [`.configu`](/interfaces/.configu) file.
```
Comment (davidmeirlevy) [inline docs/interfaces/cli/ref.mdx:16]: `--config` has been replaced, isn't it?

**Review by davidmeirlevy — COMMENTED** [review]
found one issue.

**Comment by RichardAkman** [comment]
closing to reopen so we can trigger mintilfy preview

**Pull request closed without merging**